pytroll_collectors/segments.py

```diff
@@ -851,10 +851,35 @@ def _add_existing_files_to_slot(self, slot, fnames, message):
 
 def _get_existing_files_from_message(message):
     mask = message.pattern.parser.globify({})
-    path = urlparse(message.message_data["uri"]).path
-    base_dir = os.path.dirname(path)
-
-    return glob.glob(os.path.join(base_dir, mask))
+    url_parts = urlparse(message.message_data["uri"])
+
+    if url_parts.scheme in ("", "file"):
+        base_dir = os.path.dirname(url_parts.path)
+        return glob.glob(os.path.join(base_dir, mask))
+    elif url_parts.scheme == "s3":
+        return _s3_glob(url_parts, mask)
```

Comment:
how about using just fsspec here to have a more streamlined code? eg:
```python
    return _fs_glob(url_parts, mask)
```
where `_fs_glob` is `_s3_glob` with this line on top:
```python
    fs = fsspec.filesystem(url_parts.scheme, **fsoptions)
```

Reply:
There are nothing called `fsoptions`.

Reply:
yes, this is just a suggestion in case extra parameters need to be passed.

Reply:
I think we should use fsspec for every type of filesystem here, even local.

Reply:
Done in https://github.com/pytroll/pytroll-collectors/commit/9af66a82b9aef392d5a4c3d8e2ba0636a0515a3c